File: shopica-services/src/Services/Payment/Payment.API/Models/Payment.cs
using Payment.API.Models.Enums;

namespace Payment.API.Models
{
    public class Payment : BaseEntity
    {
        public int Id { get; set; }
        public decimal Amount { get; set; }
        public DateTime PaymentDate { get; set; }
        public int PaymentMethodId { get; set; }
        public PaymentStatus paymentStatus { get; set; }
        public int CustomerId { get; set; }

        public PaymentMethod PaymentMethod { get; set; }
        public IEnumerable<Transaction> Transactions { get; set; }
    }
}

File: shopica-client/src/app/core/model/payment/payment.ts
import { Transaction } from "./transaction";

export interface Payment {
  id?: number;
  amount: number;
  paymentDate: Date,
  paymentMethodId: number;
  paymentStatus: PaymentStatus,
  customerId: number;
  transactions: Transaction[]
}

export enum PaymentStatus {
  Pending,
  Processed
}
